ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ethernet_interfaces.schema.yml

```diff
@@ -463,6 +463,53 @@ keys:
         isis_authentication_key:
           type: str
           description: Type-7 encrypted password
+        poe:
+          type: dict
+          keys:
+            disabled:
+              type: bool
+              description: Optional field used to disable PoE on a POE capable port. PoE is enabled on all ports that support it by default in EOS.
+            priority:
+              type: str
+              valid_values: ["critical", "high", "medium", "low"]
+              description: Optional field used to prioritize a port's power in the event that one of the switch's power supplies loses power
+            reboot_action:
+              type: str
+              valid_values: ["maintain", "power-off"]
+              description: Optional field used to set the PoE power behavior for a PoE port when the system is rebooted
+            link_down_action:
+              type: str
+              valid_values: ["maintain", "power-off"]
+              description: Optional field used to set the PoE power behavior for a PoE port when the port goes down
+            shutdown_action:
+              type: str
+              valid_values: ["maintain", "power-off"]
+              description: Optional field used to set the PoE power behavior for a PoE port when the port is admin down
+            limit:
+              type: dict
+              description: Optional field used to override the hardware-negotiated power limit using either wattage or a power class.
+              keys:
+                class:
```

Comment:
For classes I a not sure if this makes sense to document that it will be rendered as `classX` where `X` is the value in the CLI.

Reply:
We can perform the same mapping in AVD, so take `class` attributes, but output the same as EOS does in the generated config. We could then let a `watts` value take precedence?

Reply:
probably need to make sure the values are the same on every HW. Looking at our doc: https://www.arista.com/en/um-eos/eos-ethernet-ports#xx1313737

```
Class 0 = 15.4 W
Class 1 = 4 W
Class 2 = 7 W
Class 3 = 15.4 W
Class 4 = 30 W
Class 5 = 45 W
Class 6 = 60 W
```

VS on my vEOS device I have the same value + class7 (75W) and class8 (90W)